src/index.js

```diff
@@ -168,9 +167,7 @@ export default class ServiceWorkerPlugin {
 
     assets = validatePaths(assets, this.options)
 
-    const minify = (compiler.options.plugins || []).some(plugin => {
-      return plugin instanceof webpack.optimize.UglifyJsPlugin
-    })
+    const minify = process.env.NODE_ENV === 'production'
```

Comment:
Probably a sane default, but wouldn't it better to look at `optimization.minimize`, or introduce a plugin option? Guess this would be available under `compiler.options.optimization.minimize`

Reply:
I didn't test it but there was this warning from the official migration guide:
https://medium.com/webpack/webpack-4-migration-guide-for-plugins-loaders-20a79b927202
> Compiler
> While applying plugins Compiler.options is no longer accessible. Plugins should not access options. They should only depend on options passed to the plugin via constructor.
So I just used NODE_ENV instead

Reply:
Ok, didn't know that plugins weren't supposed to look at the global webpack options. If there's no other way to opt-in on minification based on `optimization.minimize`, the following still applies:

> Guideline: Loaders should receive all options via this.query. They should not use other ways to receive options, i. e. no property in webpack options, *no environment variable*

Brings me back to my alternative suggestion, passing it along via this.query. Using NODE_ENV to decide a default would still make sense. Unless you somehow can get access to the new `mode`, in which case it would make more sense to base defaults on that.

Reply:
That guideline was for loaders, not for plugins😄. I agree it's best to somehow access `optimization.minimize` but that doesn't seem possible for now. A dedicated option would make sense too, but lazy people like me just don't bother putting that extra config in the already-bloated `webpack.config.js` 😂

Reply:
Loaders, plugins, same difference :-D. To be honest, I was getting confused by the reference to `query`, but totally ignored that confusion. Would still imagine the note about environment variables to also count for plugins though.

Other than that, I totally agree with your reasoning. 

How about adding the option, if it's `undefined` falling back to the NODE_ENV method and emit a warning about it?